test_autofit/graphical/test_unification.py

```diff
@@ -38,3 +41,127 @@ def test_bad_id(
             AssertionError
     ):
         new_message * prior
+
+
+@pytest.fixture(
+    name="x"
+)
+def make_x():
+    return np.linspace(
+        0, 1, 100
+    )
+
+
+def test_uniform_normal(x):
+    message = UniformNormalMessage.shifted(
+        shift=1,
+        scale=2.1
+    )(
+        mean=0.0,
+        sigma=1.0
+    )
+
+    assert np.isnan(message.pdf(0.9))
+    assert np.isnan(message.pdf(3.2))
+    assert message.pdf(1.5) > 0
+
+
+def test_deferred_transform():
+    message = ShiftedMessage(
+        shift=1,
+        scale=2.1,
+        mean=0.0,
+        sigma=1.0
+    )
+
+    assert np.isnan(message.pdf(0.9))
+    assert np.isnan(message.pdf(3.2))
+    assert message.pdf(1.5) > 0
+
+
+@pytest.fixture(
+    name="message_1"
+)
+def make_message_1():
+    return ShiftedMessage(
+        shift=1,
+        scale=2.0,
+        mean=0.0,
+        sigma=1.0
+    )
+
+
+def test_values_stay_same(
+        message_1,
+):
+    assert message_1._transform.shift == 1.0
+    assert message_1._transform.scale == 2.0
+
+    message_2 = ShiftedMessage(
+        shift=2.0,
+        scale=3.0,
+        mean=0.0,
+        sigma=1.0
+    )
+    assert message_1._transform.shift == 1
+    assert message_1._transform.scale == 2.0
+
+    assert message_2._transform.shift == 2.0
+    assert message_2._transform.scale == 3.0
+
+
+@pytest.mark.parametrize(
+    "unit_value, physical_value",
+    [
+        (0.5, 2),
+        (0.0, 1),
+        (1.0, 3),
+    ]
+)
+def test_value_for(
+        message_1,
+        unit_value,
+        physical_value
+):
+    assert message_1.value_for(
+        unit_value
+    ) == pytest.approx(
+        physical_value
+    )
+
+
+@pytest.mark.parametrize(
+    "lower_limit, upper_limit, unit_value, physical_value",
+    [
+        (0.0, 1.0, 0.5, 0.5),
+        (0.0, 1.0, 1.0, 1.0),
+        (0.0, 1.0, 0.0, 0.0),
+        (1.0, 2.0, 0.5, 1.5),
+        (1.0, 2.0, 1.0, 2.0),
+        (1.0, 2.0, 0.0, 1.0),
+        (0.0, 2.0, 0.5, 1.0),
+        (0.0, 2.0, 1.0, 2.0),
+        (0.0, 2.0, 0.0, 0.0),
+    ]
+)
+def test_uniform_prior(
+        lower_limit,
+        upper_limit,
+        unit_value,
+        physical_value
+):
+    assert af.UniformPrior(
+        lower_limit=lower_limit,
+        upper_limit=upper_limit,
+    ).value_for(
+        unit_value
+    ) == pytest.approx(
+        physical_value
+    )
+
+
+def test_uniform_odd_result():
+    prior = af.UniformPrior(16, 32)
```

Comment:
This gives me inf rather than 16 - I guess because the transformed distribution is not well behaved in that domain?

@matthewghgriffiths

Reply:
is the `value_for` method returning the log probability density of the prior?

Reply:
It maps from a value in the range 0 -> 1 to a physical value. So sort of the inverse of that?

Reply:
So I think it's the percent point function (`ppf`) as used by `scipy.stats`?

I think the challenge is that for the normal distribution ppf(0) = -inf and ppf(1) = inf

Reply:
So I think for these transformed values work for the open interval (0, 1), but not for 0 or 1

Have you tried values between 0 and 1?

Reply:
it seems to still work at 0 or 1 and up to ~10, at least in terms of the limits as defined for the UniformPrior.

Those limits are used to compute shift and scale

Reply:
Might be worth a call if you have time